vm_phys.c

```diff
@@ -0,0 +1,211 @@
+#include <vm_phys.h>
+#include <queue.h>
+#include <kmem.h>
+#include <libkern.h>
+
+/* Note that generally size of page isn't constant
+ * This constant is minimal granularity of page */
+
+#define PAGE_SIZE 4096
+#define NUM_OF_PAGES 128
+#define STACK_SPACE 4 /* stack space in number of pages */
+
+#define ORDER_SIZE(x) (1 << (x))
+
+#define FAIL(MSG...) do { kprintf(MSG); while(1); } while (0);
+
+static struct vm_phys_seg basic_seg;
+static struct vm_page vm_page_array[NUM_OF_PAGES];
+
+/* Next three constants are provided by linker */
+
+/* Start of data section is start of pageable memory. */
```

Comment:
They're exposing some nasty details about the platform. Please re-read my post - I edited it.